### PR TITLE
fix: remove parameter composing logic from `avm/res/operations-management/solution` 

### PR DESCRIPTION
**Pull request by krbar**
## Description

This PR addresses the issue with some solutions, which don't follow the naming patterns described in the [template reference](https://learn.microsoft.com/en-us/azure/templates/microsoft.operationsmanagement/solutions?pivots=deployment-language-bicep). To address this, the module passes the parameters to the resource provider without attempting to compose the names. It is the responsibility of the user to provide parameters expected by the resource provider. See parameter descriptions for more details. 
Please note that this is a breaking change, the parameters of the existing deployments have to be updated to match the new format.

Related to #3378 (first part of the fix).

## Pipeline Reference

| Pipeline |
| -------- |
|       [avm.res.operations-management.solution](https://github.com/krbar/bicep-registry-modules/actions/workflows/avm.res.operations-management.solution.yml)   |

## Type of Change

- [ ] Update to CI Environment or utilities (Non-module affecting changes)
- [x] Azure Verified Module updates:
  - [ ] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [x] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [x] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] I have run `Set-AVMModule` locally to generate the supporting module files.
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

File: avm/res/operations-management/solution/main.bicep [hunk 1]
```diff
@@ -2,7 +2,16 @@ metadata name = 'Operations Management Solutions'
 metadata description = 'This module deploys an Operations Management Solution.'
 metadata owner = 'Azure/module-maintainers'
 
-@description('Required. Name of the solution. For Microsoft published gallery solution the target solution resource name will be composed as `{name}({logAnalyticsWorkspaceName})`.')
+@metadata({
+  example: '''
```
Comment (AlexanderSehr) [inline avm/res/operations-management/solution/main.bicep:6]: Why not add the example to the description so that it's part of the readme?

Comment (AlexanderSehr) [reply avm/res/operations-management/solution/main.bicep:6]: Nevermind, I see it is also in the description - which raises the opposite question: Why? 😄

File: avm/res/operations-management/solution/main.bicep [hunk 2]
```diff
@@ -11,8 +20,11 @@ param logAnalyticsWorkspaceName string
 @description('Optional. Location for all resources.')
 param location string = resourceGroup().location
 
-@description('Optional. The product of the deployed solution. For Microsoft published gallery solution it should be `OMSGallery` and the target solution resource product will be composed as `OMSGallery/{name}`. For third party solution, it can be anything. This is case sensitive.')
-param product string = 'OMSGallery'
```
Comment (AlexanderSehr) [inline avm/res/operations-management/solution/main.bicep:15]: If I see this correctly, the logic seemed to have been 'If `publisher != 'Microsoft'` the default 'OMSGallery' should be used for the `product` property. Would it make sense to keep the default, or did that default never really make much sense?

Comment (krbar) [reply avm/res/operations-management/solution/main.bicep:15]: In fact, it didn't make sense. If `publisher != 'Microsoft'`, the value must not start with `'OMSGallery'` and `publisher == 'Microsoft'`, the value was always converted to `OMSGallery/{name}`. So the default was actually never passed to the resource provider...

File: avm/res/operations-management/solution/main.bicep (hunk 1 above)
Comment (krbar) [reply avm/res/operations-management/solution/main.bicep:6]: I tried the example metadata but I couldn't locate the example in the readme. Let me check again, maybe I overlooked something...

Comment (AlexanderSehr) [reply avm/res/operations-management/solution/main.bicep:6]: That's my point. It's not a supported property 😄 I'd just keep the example in the description and remove the other

Comment (krbar) [reply avm/res/operations-management/solution/main.bicep]: Now it should be fine! :)

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**